src/System.Reflection/tests/FieldInfoTests.cs

```diff
@@ -0,0 +1,526 @@
+        [InlineData(typeof(FieldInfoTests), nameof(FieldInfoTests.stringField), false)]
+        [InlineData(typeof(FieldInfoTests), "privateIntField", false)]
+        public void IsStatic(Type type, string name, bool expected)
+        {
+            FieldInfo fieldInfo = GetField(type, name);
+            Assert.Equal(expected, fieldInfo.IsStatic);
+        }
+
+        [Theory]
+        [InlineData(typeof(FieldInfoTests), nameof(FieldInfoTests.s_assemblyField1), false)]
+        [InlineData(typeof(FieldInfoTests), nameof(FieldInfoTests.s_assemblyField2), false)]
+        [InlineData(typeof(FieldInfoTests), nameof(FieldInfoTests.s_assemblyField3), false)]
+        [InlineData(typeof(FieldInfoTests), nameof(FieldInfoTests.s_assemblyField4), false)]
+        [InlineData(typeof(FieldInfoTests), nameof(FieldInfoTests.s_assemblyField5), true)]
+        public void IsAssembly(Type type, string name, bool expected)
+        {
+            FieldInfo fieldInfo = GetField(type, name);
+            Assert.Equal(expected, fieldInfo.IsAssembly);
+        }
+
+        [Theory]
+        [InlineData(typeof(FieldInfoTests), nameof(FieldInfoTests.s_assemblyField1), false)]
+        [InlineData(typeof(FieldInfoTests), nameof(FieldInfoTests.s_assemblyField2), false)]
+        [InlineData(typeof(FieldInfoTests), nameof(FieldInfoTests.s_assemblyField3), true)]
+        [InlineData(typeof(FieldInfoTests), nameof(FieldInfoTests.s_assemblyField4), false)]
+        [InlineData(typeof(FieldInfoTests), nameof(FieldInfoTests.s_assemblyField5), false)]
+        public void IsFamily(Type type, string name, bool expected)
+        {
+            FieldInfo fieldInfo = GetField(type, name);
+            Assert.Equal(expected, fieldInfo.IsFamily);
+        }
+
+        [Theory]
+        [InlineData(typeof(FieldInfoTests), nameof(FieldInfoTests.s_assemblyField1), false)]
+        [InlineData(typeof(FieldInfoTests), nameof(FieldInfoTests.s_assemblyField2), false)]
+        [InlineData(typeof(FieldInfoTests), nameof(FieldInfoTests.s_assemblyField3), false)]
+        [InlineData(typeof(FieldInfoTests), nameof(FieldInfoTests.s_assemblyField4), false)]
+        [InlineData(typeof(FieldInfoTests), nameof(FieldInfoTests.s_assemblyField5), false)]
+        public void IsFamilyAndAssembly(Type type, string name, bool expected)
+        {
+            FieldInfo fieldInfo = GetField(type, name);
+            Assert.Equal(expected, fieldInfo.IsFamilyAndAssembly);
+        }
+
+        [Theory]
+        [InlineData(typeof(FieldInfoTests), nameof(FieldInfoTests.s_assemblyField1), false)]
+        [InlineData(typeof(FieldInfoTests), nameof(FieldInfoTests.s_assemblyField2), false)]
+        [InlineData(typeof(FieldInfoTests), nameof(FieldInfoTests.s_assemblyField3), false)]
+        [InlineData(typeof(FieldInfoTests), nameof(FieldInfoTests.s_assemblyField4), false)]
+        [InlineData(typeof(FieldInfoTests), nameof(FieldInfoTests.s_assemblyField5), false)]
```

Comment:
> Just one question: does buildtools/the build infrastructure support IL projects/IL files?

I don't believe so.  @weshaggard, it doesn't, right?  I know that one or more of the corefxlab projects uses IL rewriting to support injecting IL, but I believe that's being done with some tools checked in to corefxlab rather than part of buildtools or corefx.